services/src/contexts/mod.rs

```diff
@@ -213,6 +216,40 @@ where
     ) -> Box<dyn InitializedPlotOperator> {
         op
     }
+
+    /// This method is meant to read a ml model from disk, specified by the config key `machinelearning.model_defs_path`.
+    async fn read_ml_model(
+        &self,
+        model_sub_path: std::path::PathBuf,
+    ) -> geoengine_operators::util::Result<String> {
+        let cfg = get_config_element::<crate::util::config::MachineLearning>()
+            .map_err(|_| geoengine_operators::error::Error::InvalidMachineLearningConfig)?;
+
+        let model_base_path = cfg.model_defs_path;
+
+        let model_path = canonicalize_subpath(&model_base_path, &model_sub_path)?;
+        let model = tokio::fs::read_to_string(model_path).await?;
+
+        Ok(model)
+    }
+
+    /// This method is meant to write a ml model to disk.
+    /// The provided path for the model has to exist.
+    async fn write_ml_model(
+        &mut self,
+        model_sub_path: std::path::PathBuf,
+        ml_model_str: String,
+    ) -> geoengine_operators::util::Result<()> {
+        let cfg = get_config_element::<crate::util::config::MachineLearning>()
+            .map_err(|_| geoengine_operators::error::Error::InvalidMachineLearningConfig)?;
+
+        let model_base_path = cfg.model_defs_path;
+
+        let model_path = canonicalize_subpath(&model_base_path, &model_sub_path)?;
+        tokio::fs::write(model_path.join("model.json"), ml_model_str).await?;
```

Comment:
wouldn't it make more sense to have the filename be specified in the `model_sub_path`?

Reply:
Ok, now the method works like this:

1. check if path escapes somehow
2. try to get the parent dir of the specified file path -> create that directory if neccessary
3. create/overwrite the file

[here](https://github.com/geo-engine/geoengine/pull/639/commits/15511c1bf666395c74b1bd17f8927fcf0cf0ad1a#diff-b9fca937b91582b0887a2fea16060f5f906844f8a96587673f0f8cdf3882ef87R252-R263)